File: packages/react-native/android/src/main/java/com/breezsdkliquid/BreezSDKLiquidEventListener.kt
package com.breezsdkliquid

import breez_sdk_liquid.EventListener
import breez_sdk_liquid.SdkEvent
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter

class BreezSDKEventListener(
    private val emitter: RCTDeviceEventEmitter,
) : EventListener {
    private var id: String? = null

    fun setId(id: String) {
        this.id = id
    }

    override fun onEvent(e: SdkEvent) {
        this.id?.let {
            emitter.emit("event-$it", readableMapOf(e))
        }
    }
}

File: packages/react-native/ios/BreezSDKLiquidEventListener.swift
import Foundation
import BreezSDKLiquid

class BreezSDKEventListener: EventListener {
    private var id: String?

    func setId(id: String) {
        self.id = id
        RNBreezSDKLiquid.addSupportedEvent(name: "event-\(id)")
    }

    func onEvent(e: SdkEvent) {
        if let id = self.id {
            if RNBreezSDKLiquid.hasListeners {
                RNBreezSDKLiquid.emitter.sendEvent(withName: "event-\(id)",
                                                   body: BreezSDKLiquidMapper.dictionaryOf(sdkEvent: e))
            }
        }
    }
}
